### PR TITLE
Add CI lint stage, fix lint warnings

### PR DESCRIPTION
**Pull request by pityonline**

### CODE REVIEW

**Comment by zaquestion** [comment]
`go vet` would be a good addition here as well.

File: projects_test.go [hunk 1]
```diff
@@ -70,7 +70,7 @@ func TestListUserProjects(t *testing.T) {
 	}
 }
 
-func ListProjectsUsers_byID(t *testing.T) {
+func ListProjectsUsersByID(t *testing.T) {
```
Comment (svanharmelen) [inline projects_test.go]: If you want to fix them the right way, these functions should be called `TestList...`. And maybe (while your at it) also fix `TestGetProject_byID` and `TestGetProject_byName`?

File: .travis.yml [hunk 2]
```diff
@@ -6,6 +6,21 @@ go:
...
+stages:
+  - lint
+  - test
+
+jobs:
+  include:
+    - stage: lint
+      script:
+        - go get github.com/golang/lint/golint
+        - golint
```
Comment (svanharmelen) [inline .travis.yml]: Like @zaquestion suggested, I would add `go vet` here as well...

File: .travis.yml [hunk 3]
```diff
@@ -6,6 +6,21 @@ go:
   - 1.9.x
   - "1.10"
```
Comment (svanharmelen) [inline .travis.yml]: I know you didn't add this, but since you are tweaking this file could you also change this to `- 1.10.x` instead of `- "1.10"`? Thx!

Comment (pityonline) [reply .travis.yml]: Sure! I'll change this later.

File: projects_test.go (hunk 1 above)
Comment (pityonline) [reply projects_test.go]: Well, I found `ListProjectsUsersByXx` not tested.

```
go test -v -run ListProject
=== RUN   TestListProjectPipelines
--- PASS: TestListProjectPipelines (0.00s)
=== RUN   TestListProjects
--- PASS: TestListProjects (0.00s)
=== RUN   TestListProjectForks
--- PASS: TestListProjectForks (0.00s)
PASS
ok      github.com/xanzy/go-gitlab      0.022s
```

After I And `Test` prefix to them, a **FAIL** occurred.

```
go test -v -run ListProject
=== RUN   TestListProjectPipelines
--- PASS: TestListProjectPipelines (0.00s)
=== RUN   TestListProjects
--- PASS: TestListProjects (0.00s)
=== RUN   TestListProjectsUsersByID
--- FAIL: TestListProjectsUsersByID (0.00s)
        projects_test.go:90: Projects.ListProjectsUsers returned error: GET http://127.0.0.1:64541/projects/1/users: 404 failed to parse unexpected error type: <nil>
        projects_test.go:95: Projects.ListProjectsUsers returned [], want [0xc4200703c0 0xc420070420]
=== RUN   TestListProjectsUsersByName
--- FAIL: TestListProjectsUsersByName (0.00s)
        gitlab_test.go:37: Request url: /projects/namespace%2Fname/users?page=2&per_page=3&search=query, want /projects/namespace%2Fname/users
=== RUN   TestListProjectForks
--- PASS: TestListProjectForks (0.00s)
FAIL
exit status 1
FAIL    github.com/xanzy/go-gitlab      0.027s
```

 I'll take a look later.

**Comment by pityonline** [comment]
@zaquestion @svanharmelen According to your suggestions, I add a few more changes. Please take a look.

**Comment by svanharmelen** [comment]
I'll go ahead and merge this PR and will add the `go vet` part we talked about :wink:

Thanks @pityonline!

**Pull request merged**